gpgopt.py

```diff
@@ -0,0 +1,131 @@
+
+
+    def import_key(self, path):
+        """Check whether the gnupg pub or pvt key already exist.
+
+        If exist, import it, if not, return false.
+        """
+        try:
+            path = Path(path)
+            key_data = path.open().read()
+        except FileNotFoundError:
+            logging.error('The file path is wrong. File not found.')
+            return False
+        except OSError:
+            logging.error('Cannot read the file.')
+            return False
+        finally:
+            return False
+            pass
+
+        result = self.gpg.import_keys(key_data=key_data)
+        return result.imported, result.counts
+
+
+    def create(self, password, key_type="RSA", key_length=1024, expire_date="2y"):
+        """Create a new key pair. need parameters:
+
+        Keyword arguments:
+
+        password: required. the password of your pvt key.
+
+        key_type: optional, algorithms to generate the key.
+
+        key_length: optional, the length of key.
+
+        expire_date: optional, expire date of this key.
+        """
+        # all parameters that user can choose.
+        input_data = self.gpg.gen_key_input(name_email=self.email, key_type=key_type, passphrase=password,
+                                            key_length=key_length, expire_date=expire_date)
+
+        # generate key pair
+        return self.gpg.gen_key(input_data)
+
+
+    def export_key(self, fingerprint, export_pvt=False, password=None):
+        """Export key pair from the gnupg."""
+        sc_pub = self.gpg.export_keys(fingerprint)
+        sc_pvt = None
+
+        if export_pvt is True and password is not None:
+            sc_pvt = self.gpg.export_keys(fingerprint, True, passphrase=password)
+
+        return sc_pub, sc_pvt
+
+
+    def list_keys(self, sec=False):
+        """Get pub or pvt keys and return them."""
+        return self.gpg.list_keys(secret=sec)
+
+
+    def encrypt_message(self, message, recipient):
+        """Encrypt the message and return the encrypted message.
+
+        message: string message.
+
+        email: string, email of recipient.
+        """
+        msg = self.gpg.encrypt(message, recipient)
+        return (True, msg.data) if msg.ok else (False, msg.stderr)
+
+
+    def decrypt_message(self, messages, passphrase):
+        """Decrypt the message and return the original message.
+
+        message: list of string messages.
+
+        passphrase: string, passphrase of the private key.
+        """
+        msgs = []
+        for message in messages:
+            msg = self.gpg.decrypt(message=message, passphrase=passphrase)
+            if msg.ok is False:
+                msgs.append(msg.stderr)
+                return (False, msgs)
+            msgs.append(msg.data)
+        return (True, msgs)
+
+
+if __name__ == "__main__":
+    mygpg = GpgOpt(mygnupghome='gnupgkeys', email=getpass.getuser())
+    times = time.strftime("%Y%m%d%H%M%S",time.localtime())
```

Comment:
I would recommend making this a "constant" with a bit more descriptive name:
```suggestion
    LOG_TIME_FORMAT = time.strftime("%Y%m%d%H%M%S", time.localtime())
```

Reply:
OK.